libs/environment/src/hooks/use-environment.ts

```diff
@@ -159,7 +160,14 @@ export const useInitializeEnv = () => {
 const fetchConfig = async (url?: string) => {
   if (!url) return [];
   const res = await fetch(url);
-  const cfg = await res.json();
+  let cfg: Record<string, unknown>;
+  if (url.match(/github(.+)\.toml$/)) {
+    const content = await res.text();
+    const parsed = tomlParse(content);
+    cfg = { hosts: parsed.API.GraphQL.Hosts };
+  } else {
+    cfg = await res.json();
+  }
   const result = configSchema.parse(cfg);
```

Comment:
chore (blocking): can we update the zod schema to match what expected correct output of parsing the toml file will be. This way it will correctly throw if the underling toml file is invalid. With this change as is, parsing with zod is pointless